src/components/SwapForm/SwapActionButton/SwapOnlyButton.tsx

```diff
@@ -0,0 +1,245 @@
+      return
+    }
+
+    setProcessingSwap(true)
+
+    setBuildingRoute(true)
+    const result = await buildRoute()
+    setBuildingRoute(false)
+
+    if (result.error) {
+      setProcessingSwap(false)
+      setErrorWhileSwap(result.error)
+      return
+    }
+
+    if (!result.data?.data || !result.data?.routerAddress) {
+      setProcessingSwap(false)
+      setErrorWhileSwap(t`Build failed. Please try again`)
+      return
+    }
+
+    try {
+      await swapCallback(result.data.routerAddress, result.data.data)
+    } catch (e) {
+      setErrorWhileSwap(e.message)
+    }
+
+    setProcessingSwap(false)
+  }
+
+  const handleClickSwapForNormalMode = async () => {
+    if (!swapCallback || isBuildingRoute) {
+      return
+    }
+
+    setProcessingSwap(true)
+    // dismiss the modal will setProcessingSwap to false
+
+    setBuildingRoute(true)
+    setBuildResult(undefined)
+    const result = await buildRoute()
+    setBuildingRoute(false)
+    setBuildResult(result)
+  }
+
+  const mixpanelSwapInit = () => {
+    mixpanelHandler(MIXPANEL_TYPE.SWAP_INITIATED, {
+      gasUsd: routeSummary?.gasUsd,
+      inputAmount: routeSummary?.parsedAmountIn,
+      priceImpact: routeSummary?.priceImpact,
+    })
+  }
+
+  const handleClickSwapButton = () => {
+    mixpanelSwapInit()
+
+    setErrorWhileSwap('')
+
+    if (isAdvancedMode) {
+      handleClickSwapForAdvancedMode()
+    } else {
+      handleClickSwapForNormalMode()
+    }
+  }
+
+  const handleClickRetryForNormalMode = () => {
+    setErrorWhileSwap('')
+    handleClickSwapForNormalMode()
+  }
+
+  const renderButton = () => {
+    if (isProcessingSwap && isAdvancedMode) {
+      return (
+        <CustomPrimaryButton disabled $minimal={minimal}>
+          <Dots>
+            <Trans>Processing</Trans>
+          </Dots>
+        </CustomPrimaryButton>
+      )
+    }
+
+    if (isAdvancedMode && isSolana && !encodeSolana) {
+      return (
+        <CustomPrimaryButton disabled $minimal={minimal}>
+          <Dots>
+            <Trans>Checking accounts</Trans>
+          </Dots>
+        </CustomPrimaryButton>
+      )
+    }
+
+    if (showLoading) {
+      return (
+        <CustomPrimaryButton disabled $minimal={minimal}>
+          <Dots>
+            <Trans>Calculating</Trans>
+          </Dots>
+        </CustomPrimaryButton>
+      )
+    }
+
+    if (priceImpactResult.isVeryHigh) {
+      const shouldBeDisabled = isDisabled || (!isAdvancedMode && priceImpactResult.isVeryHigh)
+      return (
+        <CustomPrimaryButton
+          onClick={handleClickSwapButton}
+          disabled={isDisabled}
+          $minimal={minimal}
+          style={shouldBeDisabled ? undefined : { background: theme.red, color: theme.white }}
+        >
+          <Trans>Swap Anyway</Trans>
+        </CustomPrimaryButton>
+      )
+    }
+
+    if (priceImpactResult.isHigh) {
+      return (
+        <CustomPrimaryButton
+          onClick={handleClickSwapButton}
+          $minimal={minimal}
```

Comment:
What is `$minimal`?

Reply:
$minima = true if we show Approve button. There are 2 buttons so we need to make the Swap button smaller